### PR TITLE
Perf script + results

### PR DESCRIPTION
**Pull request by dsyme**
Records a few (wildly erratic) results from the compiler perf script

### CODE REVIEW

File: tests/scripts/compiler-perf.fsx [hunk 1]
```diff
@@ -82,8 +84,13 @@ let build(cloneUrl,sha:string,baseSha,ref,prNumber) =
        exec "git" ("clone " + repo + " " + dirBuild) dirBase  |> ignore
     let result = exec "git"  "reset --merge" dir
     let result = exec "git" "checkout main" dir
+<<<<<<< HEAD
```
Comment (forki) [inline tests/scripts/compiler-perf.fsx:87]: looks like merge conflict

**Comment by dsyme** [comment]
Short summary: there appear to be some minor benefits but the timings are a bit erratic (+/-10% or more)

Before:

```
build  startup-to-parseonly parseonly-to-check   check-to-nooptimize  check-to-optimize    nooptimize-to-debug
663.14 33.32                40.00                53.75                58.66                60.69               
679.69 33.60                41.28                49.24                56.43                57.23               
684.70 29.05                39.98                51.56                64.27                60.45               
```
After:
```
658.64 31.69                35.81                49.64                61.53                60.58               
616.64 33.38                33.35                49.66                62.25                61.35               
674.67 30.59                38.61                54.18                64.22                61.73               
```

**Comment by forki** [comment]
what is before and what is after?

**Comment by dsyme** [comment]
> what is before and what is after?

Before/after the recent changes to the map/set code in both FSharp.Core and compiler

**Comment by buybackoff** [comment]
I would have been a good idea to test the fat leaves impl in isolation... Is 2-cases DU with one of them trivial (Empty) optimized somehow more than 3-case DU by F# compiler?

How to run the script locally?

**Comment by cartermp** [comment]
Without a statistical benchmark these values are just noise I think. Can we contrive something up in the CompilerServiceBenchmarks project instead?

**Comment by dsyme** [comment]
> Without a statistical benchmark these values are just noise I think. 

It's hard to tell.  I've been running tess on exactly the same machine over 4 years so I can get a feel if there are massive regressions. But yes the variation is high.  It's likely major GC or OS processor speed changes kicking in at different times and may hit any perf testing we do I think

> Can we contrive something up in the CompilerServiceBenchmarks project instead?

We should yes, though I've no particular plans to myself.  One problem is making sure the code we compile is representative.

**Comment by dsyme** [comment]
@cartermp Please approve so we can at least get this update to the script merged

**Review by cartermp — APPROVED** [review]
Approved, but to be frank I don't think these numbers are meaningful as-is. The variance you mentioned indicates they aren't really an effective benchmark.

**Comment by dsyme** [comment]
> Approved, but to be frank I don't think these numbers are meaningful as-is. The variance you mentioned indicates they aren't really an effective benchmark.

Yes, but it'll catch a 50% regression, so it has its uses

**Pull request merged**